Automattic/jetpack
# Boost: Update header component

## Proposed changes:
* Convert the Header component to React

### Other information:


## Jetpack product discussion
pc9hqz-2jy-p2

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* Browse through different pages in Boost dashboard and make sure the header component is working as expected.
